src/neuroconv/tools/nwb_helpers/_configuration_models/_base_dataset_io.py

```diff
@@ -31,11 +39,20 @@ def _find_location_in_memory_nwbfile(current_location: str, neurodata_object: Co
             if isinstance(parent_field_value, dict) and neurodata_object.name in parent_field_value:
                 return parent_field_name + "/" + neurodata_object.name + "/" + current_location
         return neurodata_object.name + "/" + current_location
-    return _find_location_in_memory_nwbfile(
+    return _recursively_find_location_in_memory_nwbfile(
         current_location=neurodata_object.name + "/" + current_location, neurodata_object=parent
     )
 
 
+def _find_location_in_memory_nwbfile(neurodata_object: Container, field_name: str) -> str:
```

Comment:
Just a minor readability change to the private methods; otherwise below there was a line that read as `_find_location_in_memory_nwbfile(current_location=field_name)` which looked weird until you followed the redirection and implicit logic enough to realize how the location is constructed recursively

Reply:
I think an NWBFile can have something that's a `Container` but not an `NWBContainer`? Maybe, maybe not - the typing and logic should apply to the most generic case however